### ISSUE TITLE
Auto-update the Sphinx API documentation for Rez

### ISSUE THREAD

**Issue by ColinKennedy** [AcademySoftwareFoundation/rez#1254]
Rez currently has generated API documentation files, here: https://github.com/nerdvegas/rez/tree/master/docs/api

However it's missing many files. It'd be really great if these pages were kept up to date with each Rez release. I've done something similar to this before and here's how I think Rez could handle it.

In or before [build.py](https://github.com/nerdvegas/rez/blob/97b2841afcfb6f0d6a2c90c1b5a043e9b934c743/docs/build.py) is ran, we call `sphinx-apidoc` and give it arguments like:

`sphinx-apidoc src --output-dir docs/api`

Or if called within `build.py`, it would look like this:

```python
from sphinx.ext import apidoc

def _cli(...):
    # ... more code ...
    apidoc.main([os.path.join(REZ_SOURCE_DIR, "src"), "--output-dir", os.path.join(THIS_DIR, "api")])
    # ... more code ...
```

Then when GitHub builds the docs via https://github.com/nerdvegas/rez/blob/master/.github/workflows/sphinx.yaml#L21, the API files are added just before `sphinx-build` is ran and everything is kept up to date.

**Motivation**
In code that uses Rez API, I frequently refer to `rez.resolved_context.ResolvedContext`, `rez.packages.Package`, and `rez.packages.Variant` which are all currently there. However other additions like `rez.developer_package.DeveloperPackage` and `rez.vendor.version.version.Version` are missing. DeveloperPackage in particular is used a lot and it's a shame to not have links to it.

**Comment by ColinKennedy** [comment]
Note: an existing PR implements this feature: https://github.com/nerdvegas/rez/pull/1295

**Comment by JeanChristopheMorinPerso** [comment]
Fixed in #1522
